plugin/evm/block.go

```diff
@@ -101,15 +103,55 @@ func (b *Block) syntacticVerify() error {
 }
 
 // Verify implements the snowman.Block interface
+// Since VerifyWithContext is implemented, we never expect this to be called.
 func (b *Block) Verify(context.Context) error {
-	return b.verify(true)
+	return b.verify(nil, true)
 }
 
-func (b *Block) verify(writes bool) error {
+// ShouldVerifyWithContext implements the block.WithVerifyContext interface
+// TODO: Cache the result such that if called multiple times for the same block,
+// we do not need to recompute the value.
+func (b *Block) ShouldVerifyWithContext(context.Context) (bool, error) {
+	precompileConfigs := b.vm.currentRules().Precompiles
+	for _, tx := range b.ethBlock.Transactions() {
+		for _, accessTuple := range tx.AccessList() {
+			if _, ok := precompileConfigs[accessTuple.Address]; ok {
+				log.Debug("Should verify block with proposer VM context", "block", b.ID(), "height", b.Height())
+				return true, nil
+			}
+		}
+	}
+	log.Debug("Block does not require proposer VM context for verification.", "block", b.ID(), "height", b.Height())
+
+	return false, nil
+}
+
+// VerifyWithContext implements the block.WithVerifyContext interface
```

Comment:
could we add a type assertion for this eg,`var _ block.WithVerifyContext = &Block{}`

Reply:
Done.